tls_icon.php

```diff
@@ -39,31 +40,32 @@ function get_received_header_content($Received_Header)
 		return $Received;
 	}
 
-	public function storage_init($p)
+	public function message_load($p)
 	{
-		$headers = isset($p['fetch_headers']) ? $p['fetch_headers'] : '';
-		$p['fetch_headers'] = trim(trim($headers) . ' ' . strtoupper('Received'));
```

Comment:
Last time I checked, at least one plugin had to request the message headers so that Roundcube would fetch them. Does your approach with `get_raw_headers` not require this?

Reply:
This method will fetch the raw data from the storage instead of the available data from the hook. 

This will not only return all the headers also duplicate ones, but will also return always the same because the data is not parsed.